client/src/components/Tool/ToolForm.vue

```diff
@@ -76,7 +76,7 @@
                             <ButtonSpinner
                                 id="execute"
                                 title="Execute"
-                                @onClick="onExecute(config, currentHistory.id)"
+                                @onClick="onExecute(config, currentHistory && currentHistory.id)"
```

Comment:
```suggestion
                                @onClick="onExecute(config, currentHistory?.id)"
```

Reply:
Alternatively we could wait until the history provider resolves the current history identifier but this might cause brief delays in rendering.

Reply:
I modified it such that the tool and workflow execution forms require a valid history identifier. As mentioned this may lead to rendering delays, maybe we should consider providing an additional provider which only resolves the history id without additional metadata in order to be quick.